src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests2.cs

```diff
@@ -28872,5 +28872,787 @@ static class E
+D d = [1, 2];
+
+static class E
+{
+    extension(C c)
+    {
+        public void Add(int i) { System.Console.Write(i); }
+    }
+}
+
+public class C : System.Collections.IEnumerable
+{
+    public System.Collections.IEnumerator GetEnumerator() => null;
+}
+
+public class D : C { }
+""";
+        var comp = CreateCompilation(src);
+        CompileAndVerify(comp, expectedOutput: "12").VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void CollectionExpression_13()
+    {
+        var src = """
+C c = new C();
+D d = [.. c];
+
+static class E
+{
+    extension(C c)
+    {
+        public System.Collections.Generic.IEnumerator<int> GetEnumerator()
+        {
+            yield return 1;
+            yield return 2; 
+        }
+    }
+}
+
+public class C { }
+
+public class D : System.Collections.IEnumerable
+{
+    public System.Collections.IEnumerator GetEnumerator() => null;
+    public void Add(int i) { System.Console.Write(i); }
+}
+""";
+        var comp = CreateCompilation(src);
+        CompileAndVerify(comp, expectedOutput: "12").VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void CollectionExpression_14()
+    {
+        // Static Create extension methods does not count as a blessed Create method
+        string src = """
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+MyCollection<int> c1 = [];
+MyCollection<int> c2 = [1];
+
+[CollectionBuilder(typeof(MyCollectionBuilder), "Create")]
+class MyCollection<T>
+{
+    public IEnumerator<T> GetEnumerator() => default;
+}
+
+class MyCollectionBuilder
+{
+}
+
+static class E
+{
+    extension(MyCollectionBuilder)
+    {
+        public static MyCollection<T> Create<T>(ReadOnlySpan<T> items) => default;
+    }
+}
+""";
+        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80);
+        comp.VerifyEmitDiagnostics(
+            // (5,24): error CS9187: Could not find an accessible 'Create' method with the expected signature: a static method with a single parameter of type 'ReadOnlySpan<T>' and return type 'MyCollection<T>'.
+            // MyCollection<int> c1 = [];
+            Diagnostic(ErrorCode.ERR_CollectionBuilderAttributeMethodNotFound, "[]").WithArguments("Create", "T", "MyCollection<T>").WithLocation(5, 24),
+            // (6,24): error CS9187: Could not find an accessible 'Create' method with the expected signature: a static method with a single parameter of type 'ReadOnlySpan<T>' and return type 'MyCollection<T>'.
+            // MyCollection<int> c2 = [1];
+            Diagnostic(ErrorCode.ERR_CollectionBuilderAttributeMethodNotFound, "[1]").WithArguments("Create", "T", "MyCollection<T>").WithLocation(6, 24));
+    }
+
+    [Fact]
+    public void CollectionExpression_15()
+    {
+        // extension countable property
+        var src = """
+C c = new C();
+int[] i = [.. c];
+System.Console.Write((i[0], i[1]));
+
+static class E
+{
+    extension(C c)
+    {
+        public int Length => 2;
```

Comment:
>2 [](http://example.com/codeflow?start=29&length=1)

I assume the purpose of the test is to verify that this property isn't used. Let's throw from it then.